ci/scripts/utils/launch_java_agent.sh

```diff
@@ -1,8 +1,37 @@
-#!/bin/env bash
+#!/bin/bash
+
+# ==============================================================================
+# Script Name: launch_java_agent.sh
+#
+# Description: This script automates the process of launching a Jenkins agent
+#              on a specified machine. It ensures that the necessary
+#              prerequisites are met, such as the availability of JAVA_HOME,
+#              the Jenkins agent launch directory, and proper authentication
+#              with GitHub.
+#
+#              It then proceeds to check if the Jenkins node is online and
+#              decides whether to launch the Jenkins agent based on the node's
+#              status. The agent is launched in the background,
+#              and its PID is logged for reference.
+#
+# Prerequisites: JAVA_HOME must be set to a valid JDK installation.
+#                GitHub CLI (gh) must be installed and authenticated.
+#                Jenkins agent launch directory must exist and be specified.
+#                Jenkins controller URL and authentication token must be provided.
+#
+# Usage: ./launch_java_agent.sh [now]
+#        The optional 'now' argument forces the script to launch the Jenkins
+#        agent immediately without waiting, even if the node is detected as offline.
+#
+# ==============================================================================
+
 set -e
+shopt -s inherit_errexit
 
 controller_url="https://jenkins.epic.oarcloud.noaa.gov"
 controller_user="terry.mcguinness"
```

Comment:
Does `controller_user` need to be updated by the user running the script?  If so, can this be an input to the script with, say, `-u`?  Also, it should be noted in the documentation that this is the username for Jenkins, not the machine.

Reply:
That require it to be in the crontab too and should be clear to the user who is doing that.